packages/sitecore-jss/src/personalize/layout-personalizer.ts

```diff
@@ -0,0 +1,74 @@
+import { LayoutServiceData, ComponentRendering, HtmlElementRendering } from './../layout/models';
+
+// NULL means Hidden by this experience
+export type ComponentRenderingWithExperiences = ComponentRendering & {
+  experiences: { [name: string]: ComponentRenderingWithExperiences | null };
+};
+
+/**
+ * recursively go through all placeholders/components, check experiences nodes and replace default with object from specific experience
```

Comment:
Make this more friendly for API generated docs.
```suggestion
 * Apply personalization to layout data. This will recursively go through all placeholders/components, check experiences nodes and replace default with object from specific experience.
```